### PR TITLE
[fix][test] Catch exception when update data in mockZookeeper

### PR DESCRIPTION
**Pull request by mattisonchao**
### Motivation

Catch exception when updating data in `mockZookeeper`. this problem looks like https://github.com/apache/pulsar/pull/15755

And move `lock()` into try/catch block in case of throwing exception from the lock.
https://github.com/apache/pulsar/blob/f519e22823e80c1771716f011ecb6d693d507d2c/testmocks/src/main/java/org/apache/zookeeper/MockZooKeeper.java#L213-L226

Also fix https://github.com/apache/pulsar/issues/16483

### Modifications

- Add catch for MockZookeeper.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [x] `doc-not-needed` 
(Please explain why)
  
- [ ] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### CODE REVIEW

**Comment by mattisonchao** [comment]
The `mockZookeeper` lock may have some problems, we still need to follow up on it.

File: testmocks/src/main/java/org/apache/zookeeper/MockZooKeeper.java [hunk 1]
```diff
@@ -532,6 +540,10 @@ public void getChildren(final String path, final Watcher watcher, final Children
                 if (watcher != null) {
                     watchers.put(path, watcher);
                 }
+            } catch (Throwable ex) {
+                log.error("get children : {} error", path, ex);
+                cb.processResult(KeeperException.Code.SYSTEMERROR.intValue(), path, ctx, null);
+                return;
```
Comment (Technoboy-) [inline testmocks/src/main/java/org/apache/zookeeper/MockZooKeeper.java:546]: Better to move line 551 to `try` block and delete `return 546`

File: testmocks/src/main/java/org/apache/zookeeper/MockZooKeeper.java [hunk 2]
```diff
@@ -633,6 +645,10 @@ public void getChildren(final String path, boolean watcher, final Children2Callb
                     String child = relativePath.split("/", 2)[0];
                     children.add(child);
                 });
+            } catch (Throwable ex) {
+                log.error("get children : {} error", path, ex);
+                cb.processResult(KeeperException.Code.SYSTEMERROR.intValue(), path, ctx, null, null);
+                return;
```
Comment (Technoboy-) [inline testmocks/src/main/java/org/apache/zookeeper/MockZooKeeper.java:651]: The same with above.

**Review by Technoboy- — APPROVED** [review]
1. Change line- 991 to `Throwable`.
2. Missing to update `exists` method.

File: testmocks/src/main/java/org/apache/zookeeper/MockZooKeeper.java (hunk 2 above)
Comment (mattisonchao) [reply testmocks/src/main/java/org/apache/zookeeper/MockZooKeeper.java]: Fixed.

File: testmocks/src/main/java/org/apache/zookeeper/MockZooKeeper.java (hunk 1 above)
Comment (mattisonchao) [reply testmocks/src/main/java/org/apache/zookeeper/MockZooKeeper.java]: Fixed

**Review by codelipenghui — APPROVED** [review]

File: build/run_unit_group.sh [hunk 3]
```diff
@@ -64,7 +64,7 @@ alias echo='{ [[ $- =~ .*x.* ]] && trace_enabled=1 || trace_enabled=0; set +x; }
 
 # Test Groups  -- start --
 function test_group_broker_group_1() {
-  mvn_test -pl pulsar-broker -Dgroups='broker' -DtestReuseFork=true -DskipAfterFailureCount=1
+  mvn_test -pl pulsar-broker -Dgroups='broker' -DtestReuseFork=true -DskipAfterFailureCount=1 -DforkCount=0
```
Comment (nicoloboschi) [inline build/run_unit_group.sh:67]: this is not related to the pull. please revert

File: pom.xml [hunk 4]
```diff
@@ -100,7 +100,7 @@ flexible messaging model and an intuitive client API.</description>
       --add-opens java.management/sun.management=ALL-UNNAMED <!--JvmDefaultGCMetricsLogger-->
     </test.additional.args>
     <testReuseFork>true</testReuseFork>
-    <testForkCount>4</testForkCount>
+    <testForkCount>0</testForkCount>
```
Comment (nicoloboschi) [inline pom.xml:103]: this is not related to the pull. please revert

**Review by nicoloboschi — APPROVED** [review]
nice work! 
please revert the non related part

File: build/run_unit_group.sh (hunk 3 above)
Comment (Technoboy-) [reply build/run_unit_group.sh:67]: yes, right. We find the test failed not related to zk(may be fixed by this patch), but not reproducible locally.  So we try to modify some parameters that may affect them.

File: pom.xml (hunk 4 above)
Comment (mattisonchao) [reply pom.xml]: Fixed
